From b26aba11261ecce26600ec26d29989ef0999d28f Mon Sep 17 00:00:00 2001
From: Pavel Artsishevsky <polter.rnd@gmail.com>
Date: Sun, 11 Oct 2020 04:22:05 +0300
Subject: [PATCH] gtkentrycompletion: fix positioning for multiple monitors

Removed unneeded (and wrongly working) conditions for GtkEntryCompletion positioning.
It has been working wrong for both X and Wayland, but on Wayland the completion widget
has become completely unusable with horizontally layed out multiple monitors.

Signed-off-by: Pavel Artsishevsky <polter.rnd@gmail.com>
---
 gtk/gtkentrycompletion.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/gtk/gtkentrycompletion.c b/gtk/gtkentrycompletion.c
index ea159fe1c8..b91049c95f 100644
--- a/gtk/gtkentrycompletion.c
+++ b/gtk/gtkentrycompletion.c
@@ -1686,11 +1686,6 @@ _gtk_entry_completion_resize_popup (GtkEntryCompletion *completion)
   gtk_widget_get_preferred_size (completion->priv->popup_window,
                                  &popup_req, NULL);
 
-  if (x < area.x)
-    x = area.x;
-  else if (x + popup_req.width > area.x + area.width)
-    x = area.x + area.width - popup_req.width;
-
   if (y + entry_req.height + popup_req.height <= area.y + area.height ||
       y - area.y < (area.y + area.height) - (y + entry_req.height))
     {
-- 
2.26.2

